test/model/src/core/core.cpp

```diff
@@ -0,0 +1,138 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <algorithm>
+#include <iostream>
+
+#include "model/core.h"
+#include "wiredtiger.h"
+
+namespace model {
+
+/*
+ * NONE --
+ *     The "None" value.
+ */
+const data_value NONE = data_value::create_none();
+
+/*
+ * kv_item::~kv_item --
+ *     Delete the instance.
+ */
+kv_item::~kv_item()
+{
+    for (auto i = _updates.begin(); i != _updates.end(); i++)
+        delete *i;
```

Comment:
Yes, I thought about it—at this point, updates are just being added to the vector, and they are used outside of `kv_item` as const. This may get a little more complicated when we get to the point of removing old updates as we advance the oldest timestamp (if it becomes a part of the model), at which point ownership would be a little more complicated.